src/interpolations.jl

```diff
@@ -72,53 +115,58 @@ Return the number of base functions for an [`Interpolation`](@ref) or `Values` o
 """
 getnbasefunctions
 
-# struct that gathers all the information needed to distribute
-# dofs for a given interpolation.
-struct InterpolationInfo
-    # TODO: Can be smaller than `Int` if that matters...
-    nvertexdofs::Int
-    nedgedofs::Int
-    nfacedofs::Int
-    ncelldofs::Int
-    dim::Int
-    InterpolationInfo(interpolation::Interpolation{dim}) where {dim} =
-        new(nvertexdofs(interpolation), nedgedofs(interpolation),
-            nfacedofs(interpolation),   ncelldofs(interpolation), dim)
-end
-
 # The following functions are used to distribute the dofs. Definitions:
 #   vertexdof: dof on a "corner" of the reference shape
 #   facedof: dof in the dim-1 dimension (line in 2D, surface in 3D)
 #   edgedof: dof on a line between 2 vertices (i.e. "corners") (3D only)
 #   celldof: dof that is local to the element
 
-# Fallbacks for the interpolations which are used to distribute the dofs correctly
 """
-Number of dofs per vertex
+    vertexdof_indices(::Interpolation)
+
+Tuple of tuples containing the dof indices associated with the respective vertex.
+The indices must be consistent with the vertex indices of the reference element.
+"""
+vertexdof_indices(ip::Interpolation) = ntuple(_ -> (), nvertices(ip))
```

Comment:
As noted in #394, 👍  for giving these different names and documenting them.
To be clearer for vector-valued problems, could these be named, e.g., `vertex_refcoord_indices` (alternatively `vertex_node_indices`) since they don't refer to the local dof number, but the reference coordinate number (or dof-location).

Reply:
I am not understanding. It actually is the local dof number. Do you refer to how Ferrite implements vectorization of Lagrange interpolations?

Reply:
Yes, maybe I'm completely missing the point here?

Reply:
This (note, from master) is confusing to me: 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/1fad1afbbe5658f30ba367ffca1dc6eef12e5507/src/Dofs/DofHandler.jl#L212-L219

I.e. for each vertex dof, add `length(dh.field_dims[fi])` dofs

Reply:
Not completely missing the point, but we have a not fully functional implementation for vector-valued elements. Currently, we can only construct vector-valued elements by attaching scalar-valued bases to each dimension - which is unfortunately only part of the story. For the user-facing dof-extraction API we somehow need to compensate for this.
Further, the assumption above is in general faulty, because there are elements which are vector-valued, but cannot be constructed in such a way (e.g. Nedelec and Raviart-Thomas). I will try to fix and document this with Maxi if I find time. I think it is easier to explain when we fix the faulty assumptions first and have functional Nedelec and Raviart-Thomas elements.

Reply:
Yes, that makes sense. I think that strengthens the point regarding the function name, as what is currently extracted is the index in the `reference_coordinates(::Interpolation)` which corresponds to the dof location (with the current implementation). 
So a third option could be `vertex_dof_location_indices` (but that is getting long). If it is stated in the doc-string that it refers to the dof-location index and not the actual dof number, it is also fine. It is anyways a huge improvement that these functions are documented at all!

Reply:
Well, it *is* the local of the dof, just not for our implementation of the vector-valued Lagrange polynomials, because we vectorize the basis in at different locations in the code base (which is problematic). However, note that it is not the index of the dof location! Think about e.g. Hermite polynomials, where we have two dofs per vertex.

Reply:
I keep learning about finite elements:) - thanks! 
But since it is not in the current implementation, I would still vote for having a note about that in the docstring (until the implementation is changed).

Reply:
Sorry, I missed this comment @termi-official 
Not sure how to best write it, but a note along the lines of 

"Vector-valued interpolations are obtained by applying the same scalar `Interpolation` instance multiple times in the current implementation. For such scalar interpolations, only one degree of freedom exists per vertex. This may change in the future." 

might help clear up some confusion as to why a tuple of tuples is currently returned and what is implied by dof_index. I'm still not sure if I follow 100 % how e.g. `facedof_indices` would change if we would have a "true" vector interpolation, but I trust you know that part; so take my suggestion with a grain of salt if I've misunderstood something again. (Btw. those (facedof etc. ) devdocs are stated as missing in the preview)

I also noted that some `vertexdof_indices` returns `NTuple{N,Int}` instead of `NTuple{N,NTuple{1,Int}}`, (although that difference was not introduced by this pr):
- https://github.com/Ferrite-FEM/Ferrite.jl/blob/400ffbab36891c446bde7a6b55a5a78f614cb769/src/interpolations.jl#L424
- https://github.com/Ferrite-FEM/Ferrite.jl/blob/400ffbab36891c446bde7a6b55a5a78f614cb769/src/interpolations.jl#L361

Reply:
No problem Knut! :) Might not be super easy to explain. Please note that you can also have a scalar interpolation, but multiple dofs per vertex. My favorite example is Hermite polynomials on line elements - 4 dofs in total, 2 per vertex, each scalar-valued. I think detailed documentation should come with the PR that fixes the problem, because I do not know all code sections that will break by doing this. I think I will leave the elaboration to a future PR when the finite element spaces are better separated (see e.g. #616 and #569).

You can see this PR as the foundation to fix some problematic assumptions regarding the elements - namely that each entity has exactly the same number of dofs. Here the tuples allow e.g. to distribute 4 dofs on quad faces of a wedge and 3 dofs on triangle faces of a wedge. For a true vector-valued element, e.g. [Nedelec 1 on a triangle](https://defelement.com/elements/examples/triangle-N1curl-1.html), we would have 1 dof per face, but each call to value should return a Vec{2} and not a scalar that is vectorized internally with the help of `field_dim`. However, we would have `field_dim=2` for Nedelec 1, which currently causes several internal issues. Does this make sense? 
Further this PR unlocks the foundation for native space-time elements, which is really huge! (I am not aware of another framework that can handle this).

Also good catch, I messed up merging master. Will fix.